pkg/foundation/database/sqlite/db.go

```diff
@@ -0,0 +1,240 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/ctxutil"
+	"github.com/conduitio/conduit/pkg/foundation/database"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	_ "modernc.org/sqlite"
+)
+
+type DB struct {
+	db     *sql.DB
+	logger log.CtxLogger
+	table  string
+}
+
+var _ database.DB = (*DB)(nil)
+
+// New initializes the database structures needed by DB.
+func New(ctx context.Context, l log.CtxLogger, path, table string) (*DB, error) {
+	dbpath, err := dburl(path)
+	if err != nil {
+		return nil, cerrors.Errorf("failed to construct db path: %w", err)
+	}
+	fmt.Println("path", dbpath)
```

Comment:
Can we change this to use the logger `l.Info()...`? Also, if we do that, we should first do `l = l.WithComponent("sqlite.DB")` so the correct component is included in the output, and we can then remove that in line 66.